tests/suites/static_analysis/lint_shell.sh

```diff
@@ -0,0 +1,50 @@
+run_shellcheck() {
+  OUT=$(shellcheck --shell sh tests/*.sh tests/includes/*.sh tests/suites/**/*.sh 2>&1 || true)
+  if [ -n "${OUT}" ]; then
+    printf "\\nFound some typos"
+    echo "${OUT}"
+    exit 1
+  fi
+}
+
+run_whitespace() {
+  OUT=$(grep -Pr '\t' tests/ | grep '\.sh:' || true)
+  if [ -n "${OUT}" ]; then
+    echo "\\nERROR: mixed tabs and spaces in script: ${OUT}"
+    exit 1
+  fi
+}
+
+run_trailing_whitespace() {
+  OUT=$(grep -r " $" tests/ | grep '\.sh:' || true)
+  if [ -n "${OUT}" ]; then
+    echo "\\nERROR: trailing whitespace in script: ${OUT}"
+    exit 1
+  fi
+}
+
+test_static_analysis_shell() {
+  if [ -n "${SKIP_STATIC_SHELL:-}" ]; then
+      echo "==> SKIP: Asked to skip static shell analysis"
+      return
+  fi
+
+  (
+    set -e
+
+    cd ../
+
+    # Shell static analysis
+    if which shellcheck >/dev/null 2>&1; then
+      run "shellcheck"
+    else
+      echo "shellcheck not found, shell static analysis disabled"
+    fi
+
+    ## Mixed tabs/spaces in scripts
+    run "whitespace"
```

Comment:
I'm not sure if:
```
  run "whitespace"
```
is actually better than 
 run_whitespace

My concern is that we end up with lots of implicit abstractions in shell, and are code looks less and less like 'just bash' which is the whole point we're trying to get back to.

It might be better. We just need to be cautious around how much new test infrastructure we introduce that moves us farther and farther away from "just a bash script for easy testing".

I don't believe bash supports first-class functions (pass in the function you want to run, instead you pass in a string of the name of the function to run).
But with that, I'd probably rather that we pass in the full name of the function, rather than just a suffix of the function, because then at least "*" in Vim can find the place in the code.

Thoughts?